src/amberscript/parser.cc

```diff
@@ -1066,8 +1066,42 @@ Result Parser::ParseExpect() {
     return Result("unknown buffer name for EXPECT command");
 
   token = tokenizer_->NextToken();
-  if (!token->IsString() || token->AsString() != "IDX")
-    return Result("missing IDX in EXPECT command");
+
+  if (!token->IsString())
+    return Result("Invalid comparator in EXPECT command");
+
+  if (token->AsString() == "EQ_BUFFER") {
```

Comment:
We may refactor later to split apart if it gets to big, but I think this is fine for now.